src/theory/quantifiers/theory_quantifiers_type_rules.h

```diff
@@ -96,30 +98,30 @@ struct QuantifierInstPatternTypeRule {
   }
 };/* struct QuantifierInstPatternTypeRule */
 
-struct QuantifierInstNoPatternTypeRule {
-  inline static TypeNode computeType(NodeManager* nodeManager, TNode n, bool check)
-  {
-    Assert(n.getKind() == kind::INST_NO_PATTERN);
-    return nodeManager->instPatternType();
-  }
-};/* struct QuantifierInstNoPatternTypeRule */
-
-struct QuantifierInstAttributeTypeRule {
+struct QuantifierAnnotationTypeRule
+{
   inline static TypeNode computeType(NodeManager* nodeManager, TNode n, bool check)
   {
-    Assert(n.getKind() == kind::INST_ATTRIBUTE);
     return nodeManager->instPatternType();
   }
-};/* struct QuantifierInstAttributeTypeRule */
+}; /* struct QuantifierAnnotationTypeRule */
 
 struct QuantifierInstPatternListTypeRule {
   inline static TypeNode computeType(NodeManager* nodeManager, TNode n, bool check)
   {
     Assert(n.getKind() == kind::INST_PATTERN_LIST);
     if( check ){
-      for( int i=0; i<(int)n.getNumChildren(); i++ ){
-        if( n[i].getKind()!=kind::INST_PATTERN && n[i].getKind()!=kind::INST_NO_PATTERN && n[i].getKind()!=kind::INST_ATTRIBUTE ){
-          throw TypeCheckingExceptionPrivate(n, "argument of inst pattern list is not inst pattern");
+      for (const Node& nc : n)
+      {
+        Kind k = nc.getKind();
+        if (k != kind::INST_PATTERN && k != kind::INST_NO_PATTERN
+            && k != kind::INST_ATTRIBUTE && k != kind::INST_POOL
+            && k != kind::INST_ADD_TO_POOL && k != kind::SKOLEM_ADD_TO_POOL)
+        {
+          throw TypeCheckingExceptionPrivate(
+              n,
+              "argument of inst pattern list is not a legal quantifiers "
```

Comment:
generalize comment

Reply:
This is properly generalized currently.  Note that "INST_PATTERN_LIST" is (perhaps unfortunately) the name for a list of quantifier annotations.  Generalizing the name of this kind would be a deeper refactor that would impact the API.